### ISSUE TITLE
AWS_CONTAINER_CREDENTIALS_RELATIVE_URI not respected

### ISSUE THREAD

**Issue by hugowetterberg** [aws/aws-sdk-go-v2#105]
### Version of AWS SDK for Go?
Revision b8e7febd4b80c6517240e473410b54d4e63fed1d

### Version of Go (`go version`)?
go version go1.9.2 linux/amd64

### What issue did you see?
`external.LoadDefaultAWSConfig()` doesn't seem to respect `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`. 

Resulted in the following error when using the config:
```
EC2RoleRequestError: no EC2 instance role found caused by: RequestError: send request failed caused by: Get http://169.254.169.254/latest/meta-data/iam/security-credentials: dial tcp 169.254.169.254:80: connect: invalid argument]
```

### Steps to reproduce

Use `external.LoadDefaultAWSConfig()` in an application running in f.ex. a CodeBuild container and use the configuration in a API call.

**Comment by jasdel** [comment]
Thanks for reaching out to us @hugowetterberg, I'll try to reproduce this issue. Can you confirm that your code build environment is setup to correctly use the `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`. I think this experience can  be improved with the refactor of Config's resolvers and credential chain discussed in, #98.

**Comment by jasdel** [comment]
Thanks for pointing out this bug @hugowetterberg The issue is the SDK's `EnvConfig` type had a misspelling in its getter for that environment variable. I created #106 which fixes this bug, and adds validation checks to ensure the EnvConfig type satisfies the interface the container credentials endpoint path config resolver expects to use.

**Comment by jasdel** [comment]
Thanks for reporting this bug @hugowetterberg I merged in #106 which corrects this issue. You should now be able to use CodeBuild with container credentials.

**Comment by hugowetterberg** [comment]
Thank you for the quick response and fix @jasdel!
